lightningd/channel/channel.c

```diff
@@ -199,6 +201,11 @@ int main(int argc, char *argv[])
 		status_failed(WIRE_CHANNEL_BAD_COMMAND, "%s",
 			      tal_hex(msg, msg));
 	tal_free(msg);
+	peer->gossip_client_fd = fdpass_recv(REQ_FD);
+	if (peer->gossip_client_fd == -1)
+		status_failed(
+		    WIRE_CHANNEL_BAD_COMMAND,
+		    "Did not receive a valid client socket to gossipd");
 
```

Comment:
fds needed at startup are more easily passed directly on the new_subd() call, so I've been using that where possible.  This can be done as a follow-up patch though.